src/gui.c

```diff
@@ -81,6 +91,27 @@ uzbl_gui_init ()
     } else {
         window_init ();
     }
+
+    uzbl.gui_->im_context = gtk_im_context_simple_new ();
+    gtk_im_context_reset (uzbl.gui_->im_context);
+
+    g_signal_connect (uzbl.gui_->im_context, "commit",
+        G_CALLBACK (uzbl_commit_cb), 0);
+}
+
+static void
+uzbl_commit_cb (
+ GtkIMContext* context,
+ const gchar *str,
+ void* dummy)
+{
+    (void)context;
+    (void)dummy;
+
+    uzbl_events_send (KEY_PRESS, NULL,
+        TYPE_STR, "",
```

Comment:
What if we copy the state (held modifiers) in every keypress/keyrelease to the user_data of this function. See send_keypress_event for this stuff involved for this.